rpcs3/Emu/Io/Turntable.cpp

```diff
@@ -0,0 +1,301 @@
+}
+
+usb_device_turntable::~usb_device_turntable()
+{
+}
+
+void usb_device_turntable::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue, u16 wIndex, u16 wLength, u32 buf_size, u8* buf, UsbTransfer* transfer)
+{
+	transfer->fake = true;
+
+	// Control transfers are nearly instant
+	switch (bmRequestType)
+	{
+		case 0x21:
+			switch (bRequest)
+			{
+				case 0x09:
+					// Do nothing here - not sure what it should do.
+					break;
+				default:
+			        turntable_log.error("Unhandled Query Type: 0x%02X", buf[0]);
+					break;
+			}
+			break;
+		default:
+			usb_device_emulated::control_transfer(bmRequestType, bRequest, wValue, wIndex, wLength, buf_size, buf, transfer);
+			break;
+	}
+}
+
+void usb_device_turntable::interrupt_transfer(u32 buf_size, u8* buf, u32 /*endpoint*/, UsbTransfer* transfer)
+{
+	transfer->fake            = true;
+	transfer->expected_count  = buf_size;
+	transfer->expected_result = HC_CC_NOERR;
+	// Interrupt transfers are slow
+	// Turntable runs at 100hz --> 10ms
+	transfer->expected_time = get_timestamp() + 10000;
+
+	memset(buf, 0, buf_size);
+
+	buf[0] = 0x00; // Face Buttons
+	// FACE BUTTON HEXMASK:
+	// 0x01 = Square
+	// 0x02 = Cross
+	// 0x04 = Circle
+	// 0x08 = Triangle / Euphoria
+
+	buf[1] = 0x00; // Start/Select Buttons
+	// START/SELECT HEXMASK:
+	// 0x01 = Select
+	// 0x02 = Start
+	// 0x10 = PS Button
+
+	buf[2] = 0x0F; // D-Pad
+	// DPAD VALUES:
+	// 0x00 = Up
+	// 0x01 = Up-Right
+	// 0x02 = Right
+	// 0x03 = Right-Down
+	// 0x04 = Down
+	// 0x05 = Down-Left
+	// 0x06 = Left
+	// 0x07 = Up-Left
+	// 0x0F = None
+
+	buf[3] = 0x80; // Unknown, always 0x80
+	buf[4] = 0x80; // Unknown, always 0x80
+
+	buf[5] = 0x80; // Left Turntable
+	buf[6] = 0x80; // Right Turntable
+
+	// The following bytes are NOTed (set to 0xFF) when active.
+	// If multiple buttons are pressed for one byte, the byte is NOTed twice (reset to 0x00).
+	buf[7]  = 0x00; // Square Button / D-Pad Right
+	buf[8]  = 0x00; // D-Pad Left
+	buf[9]  = 0x00; // Cross Button / D-Pad Up
+	buf[10] = 0x00; // D-Pad Down
+	buf[11] = 0x00; // Triangle / Euphoria Button
+	buf[12] = 0x00; // Circle Button
+
+	buf[19] = 0x00; // Effects Dial, lower 8 bits
+	buf[20] = 0x02; // Effects Dial, upper 2 bits
+
+	buf[21] = 0x00; // Crossfader, lower 8 bits
+	buf[22] = 0x02; // Crossfader, upper 2 bits
+
+	buf[23] = 0x00; // Platter Buttons
+	// PLATTER BUTTON VALUES:
+	// 0x01 = Right Platter, Green
+	// 0x02 = Right Platter, Red
+	// 0x04 = Right Platter, Blue
+	// 0x10 = Left Platter, Green
+	// 0x20 = Left Platter, Red
+	// 0x40 = Left Platter, Blue
+
+	buf[24] = 0x02; // Unknown, always 0x02
+	buf[26] = 0x02; // Unknown, always 0x02
+
+	// All other bufs are always 0x00
+
+	const auto handler = pad::get_current_handler();
+	const auto& pads   = handler->GetPads();
+	const auto pad     = pads[6];
+
+	if (!(pad->m_port_status & CELL_PAD_STATUS_CONNECTED))
+		return;
+
+	for (Button& button : pad->m_buttons)
```

Comment:
I don't see you modifying the button. So const should be more fitting
```suggestion
	for (const Button& button : pad->m_buttons)
```